### DISCUSSION TITLE
Hva konkret betyr validering og kontrollering av attributtverdier?

### DISCUSSION THREAD

**Discussion by troelde** [NorskHelsenett/Tillitsrammeverk#47] [General]
Ref.6.1 Cybersikkerhet
...
_Informasjonen i datamodellen vil blant annet benyttes til å utføre analyse av logger, og vil kunne bli vist til sluttbrukere i forskjellige applikasjoner. Dette åpner for angrep mot sårbarheter i programvare, som f.eks. misbruk av makroer eller XSS angrep i nettlesere. Sannsynligheten for denne typen sikkerhetsangrep bør begrenses ved **at verdier som overføres kan valideres og kontrolleres**._

Tenker det er greit å ta en runde på hva konkret _validering_ og _kontrollering_ faktisk innebærer. Når vi ser på [attributtmodellen til HSØ](https://github.com/HL7Norway/AuditEvent/blob/main/attributtliste.md), så er det jo lagt inn en fleksibilitet mht. hvilke kodeverk som brukes for hver av attributtene, ref. angivelse av `:system` (og `:assigner`) for hver av attributtene.

**Comment by richardhus** [comment]
Støtter deg i dette @troelde, jeg tenker at validering og kontroll i første omgang bør avgrenses til om innholdet som formidles mellom konsumenter og kilder ser ut til å inneholder tegn som kan medføre fare for angrep. Avgrenser vi innhold i lokale kodeverk og/eller fritekstfelter til et definert maksimalt antall av ufarlige tegn som 0-9, a-å, A-Å og kanskje komma, punktum og eventuelt om det ikke er risiko heller ved bindestrek/tanskestrek, skråstrek, spørsmålstegn, pluss-tegn så bør det vel være relativt små muligheter for denne typen angrep?

Jeg kan for lite om dette til å være skråsikker her, men vi må også ta i betraktning at det finnes en tilsvarende trussel (antakelig større) ved å formidle og vise innholdet i fritekstdokumenter (altså det hele løsningen med dokumentdeling handler om) til alle konsumenter som bruker diverse nettlesere lokalt for det... Innhold fra rene tekst-felt som EPJ-leverandører ikke tillater allverdens tegn i fra før av, der flesteparten kun er tilgjengelig for redigering av leverandøren eller priviligerte brukere hos konsumentene utgjør forhåpentligvis ikke den viktigste angrepsvektoren å være bekymret for...

Tilsvarende som for fritekstdokumenter kan også gjelde mer strukturert informasjon (datadeling) der det neppe blir bare kodeverk som skal formidles der heller, fritekst og diverse MTU-er som dytter inn mye rart av data er antakelig vel så skumle som det lille og ganske kontrollerte innholdet vi skal formidle i sikkerhetsbevisene. Men krav til EPJ-leverandører som skal gå god for det som formidles til HelseID som i størst mulig grad reduserer mulig risiko (uten å gå på akkord med behovene som skal dekkes - dette må være forholdsmessige tiltak) er jeg for altså.
